app/userAuthentication.js

```diff
@@ -0,0 +1,59 @@
+import fetch from "electron-fetch/lib"
+import appInfo from "./app-info.json"
+import keytar from "keytar"
+import {BrowserWindow, session} from "electron"
+
+let mainWindow, authWindow
+
+const clientId = appInfo["client_id"]
+const clientSecret = appInfo["client_secret"]
+const requiredScopes = ["repo"].join("%20")
+
+export function authorizeUser (mainWindowRef) {
+  mainWindow = mainWindowRef
+  openAuthWindow()
+}
+
+export function fetchAccessToken (code) {
+  authWindow.destroy()
+  var data = {
+    client_id: clientId,
+    client_secret: clientSecret,
+    code: code,
+  }
+  fetch("https://github.com/login/oauth/access_token", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json; charset=utf-8",
+      "Accept": "application/json",
+    },
+    body: JSON.stringify(data),
+  })
+    .then(response => response.json())
+    .then(data => {
+      keytar.setPassword("Classroom-Desktop", "token", data.access_token)
+      mainWindow.webContents.send("receivedAuthorization")
+    })
+    .catch((error) => console.log(error))
```

Comment:
:art: If something fails with this flow you might want to show some UI to the user - this could be done with IPC, but I wouldn't worry about it in the scope of this PR.